src/test/java/uk/ac/ebi/ampt2d/accession/commons/core/BasicAccessioningServiceSaveDelegateTest.java

```diff
@@ -0,0 +1,106 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.accession.commons.core;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.annotation.Commit;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.transaction.TestTransaction;
+import uk.ac.ebi.ampt2d.accession.commons.generators.SingleAccessionGenerator;
+import uk.ac.ebi.ampt2d.accession.commons.generators.exceptions.AccessionCouldNotBeGeneratedException;
+import uk.ac.ebi.ampt2d.accession.commons.hashing.SHA1HashingFunction;
+import uk.ac.ebi.ampt2d.accession.commons.persistence.BasicSpringDataRepositoryDatabaseService;
+import uk.ac.ebi.ampt2d.test.TestModel;
+import uk.ac.ebi.ampt2d.test.configuration.TestDatabaseServiceTestConfiguration;
+import uk.ac.ebi.ampt2d.test.persistence.TestEntity;
+import uk.ac.ebi.ampt2d.test.persistence.TestRepository;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@ContextConfiguration(classes = {TestDatabaseServiceTestConfiguration.class})
+public class BasicAccessioningServiceSaveDelegateTest {
+
+    @Autowired
+    private TestRepository repository;
+
+    @Autowired
+    private BasicSpringDataRepositoryDatabaseService<TestModel, TestEntity, String, String> databaseService;
+
+    @Test
+    public void testUpdateWhenHashAndAccessionCoincide() throws AccessionCouldNotBeGeneratedException {
+        repository.save(new TestEntity("id-test-3", "3323D8A8F66A0602CC59372E866DD8E116DCCDB2", "test-3"));
+        TestTransaction.end();
+
+        Map<String, TestModel> accessions = new HashMap<>();
+        accessions.put("3323D8A8F66A0602CC59372E866DD8E116DCCDB4", TestModel.of("test-1"));
+        accessions.put("3323D8A8F66A0602CC59372E866DD8E116DCCDB3", TestModel.of("test-2"));
+        accessions.put("3323D8A8F66A0602CC59372E866DD8E116DCCDB2", TestModel.of("test-3"));
+
+        BasicAccessioningServiceSaveDelegate<TestModel, String, String> delegate =
+                new BasicAccessioningServiceSaveDelegate<>(databaseService);
+
+        delegate.doSaveAccessions(getAccessioningService().getAccessionGenerator().generateAccessions(accessions));
+
+        TestTransaction.start();
+        TestTransaction.flagForCommit();
+        repository.delete("id-test-3");
+        TestTransaction.end();
+    }
+
+    @Test
+    @Commit
+    public void testDoNotAccessionWhenHashIsEqualButIdDifferent() throws AccessionCouldNotBeGeneratedException {
```

Comment:
Again, I would expect something being asserted here (Object X does not get accession), but I don't see where is this being done.